### PR TITLE
multimap.rebuild implemented on eo

### PR DESCRIPTION
**Pull request by EugeneDar**
https://github.com/objectionary/eo-collections/issues/15

### CODE REVIEW

**Comment by EugeneDar** [comment]
@Graur @levBagryansky please have a look

**Comment by EugeneDar** [comment]
@Graur @levBagryansky please have a look

File: src/main/eo/org/eolang/collections/multimap.eo [hunk 1]
```diff
@@ -111,6 +111,81 @@
           a
           a.with x
 
+  [harr arr] > rebuilded
```
Comment (levBagryansky) [inline src/main/eo/org/eolang/collections/multimap.eo:114]: @EugeneDar I think it would be better to add description what the object does

**Review by levBagryansky — CHANGES_REQUESTED** [review]
@EugeneDar One comment below

File: src/test/eo/org/eolang/collections/multimap-tests.eo [hunk 2]
```diff
@@ -113,14 +120,15 @@
       list
         * "c" 0 "a" "c"
 
+# This test has been temporarily disabled
```
Comment (Graur) [inline src/test/eo/org/eolang/collections/multimap-tests.eo:123]: @EugeneDar We can add a pdd puzzle here

File: src/test/eo/org/eolang/collections/multimap-tests.eo [hunk 3]
```diff
@@ -129,25 +137,29 @@
...
   mp.without 1 > mp2!
   assert-that > @
-    * (mp.size) (mp2.size)
-    $.equal-to
-      list
-        * 3 1
+    mp2.size
+    $.equal-to 1
+
+# This test has been temporarily disabled
```
Comment (Graur) [inline src/test/eo/org/eolang/collections/multimap-tests.eo:154]: @EugeneDar Same here. We can add a pdd puzzle

**Review by Graur — CHANGES_REQUESTED** [review]

File: src/main/eo/org/eolang/collections/multimap.eo [hunk 4]
```diff
@@ -111,6 +111,81 @@
...
+    mappedi. > wrapped-pairs!
```
Comment (Graur) [inline src/main/eo/org/eolang/collections/multimap.eo:115]: @EugeneDar Maybe we can rename this object? Something short, like `pairs`, `pairs-list` etc

**Review by Graur — CHANGES_REQUESTED** [review]
@EugeneDar I think we need to refactor object' names and make some of them shorter

File: src/test/eo/org/eolang/collections/multimap-tests.eo (hunk 2 above)
Comment (EugeneDar) [reply src/test/eo/org/eolang/collections/multimap-tests.eo]: @Graur fixed it.

File: src/test/eo/org/eolang/collections/multimap-tests.eo (hunk 3 above)
Comment (EugeneDar) [reply src/test/eo/org/eolang/collections/multimap-tests.eo]: @Graur fixed it.

File: src/main/eo/org/eolang/collections/multimap.eo (hunk 4 above)
Comment (EugeneDar) [reply src/main/eo/org/eolang/collections/multimap.eo:117]: @Graur I agree, I think `sorted-pairs` will be better.

File: src/main/eo/org/eolang/collections/multimap.eo [hunk 5]
```diff
@@ -111,6 +111,81 @@
...
+      list
+        arr
+      [x i]
+        mod. > new-index!
+          number
+            (harr.at i)
+          (arr.length)
+        comparable-pair (* new-index x) > @
+    sorted. > sorted-pairs!
```
[the remark this answers is not in the document]
Comment (EugeneDar) [reply src/main/eo/org/eolang/collections/multimap.eo:126]: @Graur I think it will decrease code complexity and increase readability.

**Comment by EugeneDar** [comment]
@levBagryansky please have a look again

**Review by Graur — APPROVED** [review]
@EugeneDar Thanks!

**Comment by yegor256** [comment]
@rultor merge

**Pull request merged**